### PR TITLE
terraform.limits

### PR DESCRIPTION
**Pull request by igorkotof**
Find ability to estimate if all there are any limits that we would need to raise to deploy infra

**Testing:**
[asciicast](https://asciinema.org/a/QVkX30tP7GUkLb778VfbTm1Qx)

### CODE REVIEW

File: terraform/main.mk [hunk 1]
```diff
@@ -7,7 +7,23 @@ ENV_DIR ?= $(INFRA_DIR)/env/$(ENV)
 OUTPUT_JSON_FILE = $(INFRA_DIR)/env/$(ENV)/output.json
 TERRAFORM_VERSION ?= "0.12.29"
 
-
+AWS_LIMITS_LIST ?= $$(echo "{ \
```
Comment (AutomationD) [inline terraform/main.mk:10]: @igorkotof Let's add a quick comment or link to a documentation to describe what it does

File: terraform/main.mk [hunk 2]
```diff
@@ -7,7 +7,23 @@ ENV_DIR ?= $(INFRA_DIR)/env/$(ENV)
...
+\"aws_eip\":[ \
+	{\"value\":\"5\", \"service\":\"vpc\", \"quotacode\":\"L-2AFB9258\"}] \
+}")
+
+AWS_LIMITS ?= @ ( echo $(foreach item, $(shell echo $(AWS_LIMITS_LIST) | $(JQ) -e -r '. | to_entries[] | .key' ), \
+$(shell echo $$(if [ $(shell grep -c "+ resource \"$(item)\"" $(ENV_DIR)/tfplan.txt) -lt $(shell echo $(AWS_LIMITS_LIST) | $(JQ) -r '.$(item)[].value') ]; \
+then echo "\n\033[32m[OK]\033[0m $(item) limit"; \
+else echo "\n\033[33m[WARNING]\033[0m $(item) limit \(Value:$(shell echo $(AWS_LIMITS_LIST) | $(JQ) -r '.$(item)[].value')\) exceeded! \
+Current value:$(shell grep -c "+ resource \"$(item)\"" $(ENV_DIR)/tfplan.txt) \
+\033[33m To request a service quota increase:\033[0m \033[36m aws service-quotas request-service-quota-increase --service-code $(shell echo $(AWS_LIMITS_LIST) | $(JQ) -r '.$(item)[].service') --quota-code $(shell echo $(AWS_LIMITS_LIST) | $(JQ) -r '.$(item)[].quotacode') --desired-value \<your_desired_value\> \033[0m"; fi ))) )
```
Comment (AutomationD) [inline terraform/main.mk:26]: Let's add another newline before the actual commands list

**Review by AutomationD — CHANGES_REQUESTED** [review]
Looks good, with minor changes

File: terraform/main.mk (hunk 2 above)
Comment (igorkotof) [reply terraform/main.mk:26]: @AutomationD you know yesterday I spent time to try to format output with new lines but due to the `echo` at the begining of `AWS_LIMITS` command it doesn't work, any ideas how to make output info with new lines?

File: terraform/main.mk (hunk 1 above)
Comment (igorkotof) [reply terraform/main.mk:13]: done

File: terraform/main.mk (hunk 2 above)
Comment (AutomationD) [reply terraform/main.mk:29]: try `\n` ?

Comment (igorkotof) [reply terraform/main.mk:29]: @AutomationD I tried - no result, also tried to replace `echo` with `printf` - didn't work as well :(

Comment (igorkotof) [reply terraform/main.mk]: @AutomationD Output fixed, please approve
image

**Review by AutomationD — APPROVED** [review]
Looks good!

**Pull request merged**